src/foreign/core/num.rs

```diff
@@ -94,7 +94,7 @@ macro_rules! implement_nonzero_int {
     ($nonzero:ty, $int:ty) => {
         impl<'a> Arbitrary<'a> for $nonzero {
             fn arbitrary(u: &mut Unstructured<'a>) -> Result<Self> {
-                match Self::new(<$int as Arbitrary<'a>>::arbitrary(u)?) {
+                match Self::new(<$int as Arbitrary<'a>>::arbitrary(u)?.saturating_add(1)) {
                     Some(n) => Ok(n),
                     None => Err(Error::IncorrectFormat),
```

Comment:
It is kind of nice that `1` becomes `NonZero(1)`, etc...

How about we change the `None => ...` arm to return `Self::new($int::MAX).unwrap()`?

Reply:
I don't think it makes any difference. But I guess there's some intangible niceness about that. I can change it.